File: src/main/java/dev/lampirg/consultationappointment/data/appointment/validator/InTimeBorders.java
package dev.lampirg.consultationappointment.data.appointment.validator;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Constraint(validatedBy = InTimeBordersValidator.class)
public @interface InTimeBorders {
    String message() default "Invalid time borders";
    Class<?>[] groups() default {};
    Class<? extends Payload>[] payload() default {};
}

File: src/main/kotlin/dev/lampirg/consultationappointment/data/appointment/validator/InTimeBorders.kt
package dev.lampirg.consultationappointment.data.appointment.validator

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [InTimeBordersValidator::class])
annotation class InTimeBorders(
    val message: String = "Invalid time borders",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)